fix: Return shallow copy of Metadata

Cobalt0s opened the pull request:
# Description
Noticed that a test would fail under parallel execution: https://github.com/amp-labs/connectors/actions/runs/12360650495/job/34496076253?pr=1220#step:5:1728

# Fix
Located an issue shared by connectors that create `*common.ListObjectMetadataResult` relying on shared static schema. A copy of the map is now used to construct object metadata.

Cobalt0s commented:
* **#1221** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1221?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1221?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Cobalt0s commented:
I am not sure why the linter started complaining. Running locally the correct version without any reported issues.
```
golangci-lint --version
golangci-lint has version v1.62.0 built with go1.23.3 from (unknown, modified: ?, mod sum: "h1:/G0g+bi1BhmGJqLdNQkKBWjcim8HjOPc4tsKuHDOhcI=") on (unknown)
```
Tried to download tar with this program and it also doesn't give an error (I ran it with cleared cache).
[Linter logs](https://github.com/amp-labs/connectors/actions/runs/12362841095/job/34502998593?pr=1221) (screenshot below)
![image](https://github.com/user-attachments/assets/aa5a8b50-b883-4b9c-8e52-0dcc09794402)

@eberle1080

eberle1080 commented:
There must be a github bug. Either that or your branch is way out of date. Check `.golangci.yml` in the root directory, it's already set to `colored-line-number`.

eberle1080 commented:
FWIW I'm suddenly having failures as well, unrelated to my PR. See https://github.com/amp-labs/connectors/pull/1222. Like you, I can't repro these issues locally.

Cobalt0s commented:
> FWIW I'm suddenly having failures as well, unrelated to my PR. See #1222. Like you, I can't repro these issues locally.

Ok, so I added nolinter on the line that Github action is suggesting (I usually overlook this, but it was right there, however locally I dont get this message and the code of issue is prexisting).
![image](https://github.com/user-attachments/assets/638af9ef-d9a7-42cf-9fd4-7b54383d75c8)

Anyway, this PR has a valid lint action now. Once this is merged all the test issues that you have on your PR and also on main would disappear as I fixed the bug. (The implecations of this bug I assume are not critical, but it was wrong nonetheless).

@eberle1080
